src/math/power.c

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2021 Intel Corporation. All rights reserved.
+//
+// Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+//
+//
+
+ /* Include Files */
+#include <sof/math/power.h>
+
+/* Function Declarations */
+static int32_t div_s32s64(int64_t denominator);
+
+/* Function Definitions */
+/**
+ * Arguments	: int64_t denominator
+ *
+ * Return Type	: int32_t
+ */
+static inline int32_t div_s32s64(int64_t denominator)
+{
+	uint64_t b_denominator;
+	int32_t quotient;
+
+	if (denominator == 0) {
+		quotient = INT32_MAX;
+	} else {
+		if (denominator < 0)
+			b_denominator = ~(uint64_t)denominator + 1;
+		else
+			b_denominator = (uint64_t)denominator;
+
+		b_denominator = 1125899906842624ULL / b_denominator;
+		if (denominator < 0L)
+			quotient = (int32_t)-(int64_t)b_denominator;
+		else
+			quotient = (int32_t)b_denominator;
+	}
+	return quotient;
+}
+
+/**
+ * Arguments	: int32_t b
+ * base = [-1 to 2^5];Q7.25
```

Comment:
The value range of Q7.25 should be (-2^6, 2^6]

Reply:
base range is `[-32 to +32]` i.e.` Q6.26` , `[-2^5 , 2^5]`

Reply:
Sorry I didn't get your point. The format you mentioned in the comment is Q7.25, why you use Q6.26 to get the value range?

Reply:
`Q7.25` is wrong comment. I will update with `Q.6.26`.  Power function algorithm can  handle all possible range between `-/+32^[-/+1 to -/+3]  ` .